internal/agent/device/config/controller.go

```diff
@@ -1,91 +1,130 @@
 package config
 
 import (
+	"context"
 	"fmt"
 
+	ignv3types "github.com/coreos/ignition/v2/config/v3_4/types"
 	"github.com/flightctl/flightctl/api/v1alpha1"
 	"github.com/flightctl/flightctl/internal/agent/device/fileio"
+	"github.com/flightctl/flightctl/internal/agent/device/hook"
 	"github.com/flightctl/flightctl/pkg/log"
+	"github.com/samber/lo"
 )
 
 // Config controller is responsible for ensuring the device configuration is reconciled
 // against the device spec.
 type Controller struct {
-	deviceWriter *fileio.Writer
-	trackedFiles map[string]struct{}
+	hookManager  hook.Manager
+	deviceWriter fileio.Writer
 	log          *log.PrefixLogger
 }
 
 // NewController creates a new config controller.
 func NewController(
-	deviceWriter *fileio.Writer,
+	hookManager hook.Manager,
+	deviceWriter fileio.Writer,
 	log *log.PrefixLogger,
 ) *Controller {
 	return &Controller{
+		hookManager:  hookManager,
 		deviceWriter: deviceWriter,
-		trackedFiles: make(map[string]struct{}),
 		log:          log,
 	}
 }
 
-func (c *Controller) Sync(current *v1alpha1.RenderedDeviceSpec, desired *v1alpha1.RenderedDeviceSpec) error {
-	c.log.Debug("Syncing device configuration")
-	defer c.log.Debug("Finished syncing device configuration")
+func (c *Controller) Sync(ctx context.Context, current, desired *v1alpha1.RenderedDeviceSpec) error {
+	c.log.Info("Syncing device configuration")
+	defer c.log.Info("Finished syncing device configuration")
 
-	if desired.Config == nil {
-		c.log.Debug("Device config is nil")
-		return nil
+	if err := c.hookManager.Sync(current, desired); err != nil {
```

Comment:
we should probably move this into its own hooks controller, hooks are a top level spec.

Reply:
The hookManager doesn't have periodic functionality besides the `Sync` operation. So we can call it from topLevel.
Is there any additional functionality that needs to be moved to hooks controller?
The callback are still called from the controller.

Reply:
Moved the manager also to the top.  Sync is done from there.

Reply:
ok for now